tests/feinterface/fe_interface_view_get_function_from_local_dof_values_01.cc

```diff
@@ -0,0 +1,467 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Check that FEInterfaceViews::get_function_*_from_local_dof_values
+// works with different number types.
+
+#include <deal.II/base/quadrature_lib.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_renumbering.h>
+
+#include <deal.II/fe/fe_dgp.h>
+#include <deal.II/fe/fe_interface_values.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/vector.h>
+
+#include <fstream>
+#include <type_traits>
+
+#include "../tests.h"
+
+template <typename NumberType, int dim, typename ExtractorType>
+void
+test_feiv_view(const Vector<double> &         solution,
+               const FEInterfaceValues<dim> & fe_iv,
+               const unsigned int &           n_q_points,
+               const ExtractorType &          extractor,
+               const std::vector<NumberType> &local_dof_values);
+
+// Scalar view
+template <typename NumberType, int dim>
+void
+test_feiv_view(const Vector<double> &            solution,
+               const FEInterfaceValues<dim> &    fe_iv,
+               const unsigned int &              n_q_points,
+               const FEValuesExtractors::Scalar &extractor,
+               const std::vector<NumberType> &   local_dof_values)
+{
+  using View = typename std::remove_reference<
+    typename std::remove_const<decltype(fe_iv[extractor])>::type>::type;
```

Comment:
You need not change this, but FYI the `std::remove_reference<std::remove_const<T>>` is the same as a `std::decay<T>`.

Reply:
good know, thanks!